Sources/NIO/RecvByteBufferAllocator.swift

```diff
@@ -89,78 +66,50 @@ public struct AdaptiveRecvByteBufferAllocator: RecvByteBufferAllocator {
         precondition(initial > minimum, "initial: \(initial)")
         precondition(maximum > initial, "maximum: \(maximum)")
 
-        let minIndex = AdaptiveRecvByteBufferAllocator.sizeTableIndex(minimum)
-        if AdaptiveRecvByteBufferAllocator.sizeTable[minIndex] < minimum {
-            self.minIndex = minIndex + 1
-        } else {
-            self.minIndex = minIndex
-        }
-
-        let maxIndex = AdaptiveRecvByteBufferAllocator.sizeTableIndex(maximum)
-        if AdaptiveRecvByteBufferAllocator.sizeTable[maxIndex] > maximum {
-            self.maxIndex = maxIndex - 1
-        } else {
-            self.maxIndex = maxIndex
-        }
-
-        self.initial = initial
-        self.minimum = minimum
-        self.maximum = maximum
+        // We need to round all of these numbers to a power of 2. Initial will be rounded down,
+        // minimum down, and maximum up.
+        self.minimum = minimum.previousPowerOf2()
```

Comment:
Do we need to `min` this with the max allocation size here as well? Looks like we could end up with `minimum > initial` if minimum and initial were sufficiently large

Reply:
Yes, we do. Good catch.